nat/nat.go

```diff
@@ -150,6 +150,10 @@ type PortMapping struct {
 	Binding PortBinding
 }
 
+func (p *PortMapping) String() string {
+	return fmt.Sprintf("%s:%s:%s", p.Binding.HostIP, p.Binding.HostPort, p.Port)
```

Comment:
We should use `net.JoinHostPort()` here, to account for IPv6 addresses; https://pkg.go.dev/net#JoinHostPort

e.g.;

```suggestion
	return net.JoinHostPort(p.Binding.HostIP, p.Binding.HostPort+":"+p.Port.Port())
```

But I need to have a closer look, as there's many possible formats (ports mapped here can also have the protocol (e.g. `80/tcp`), which may have to be taken into account here.

And of course we also have `PortMap` (which is a list of mapped ports) 😞

Reply:
Oh, I guess `HostPort` doesn't have the scheme, so maybe that part is not an issue

Reply:
Do we know if `HostIP` defaults to `0.0.0.0` (or `::` IPv6)? 🤔